themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -88,147 +89,143 @@ VuFind.register('itemStatuses', function ItemStatuses() {
+          $.ajax({
+            // todo: replace with fetch
+            url: VuFind.path + url,
+            data: { id: items.map((item) => item.id) },
+            dataType,
+            method,
+          })
+            .done(done)
+            .catch(error);
+        });
+      },
+      success: itemStatusAjaxSuccess,
+      failure: itemStatusAjaxFailure,
+      delay,
+    });
+  }
 
   //store the handlers in a "hash" obj
   var checkItemHandlers = {
-    'ils': ItemStatusHandler,
-    'overdrive': OdItemStatusHandler,
+    ils: makeItemStatusQueue(),
+    overdrive: makeItemStatusQueue({ url: "/Overdrive/getStatus" }),
   };
 
   function checkItemStatus(el) {
-    var $item = $(el);
-    if ($item.hasClass('js-item-pending') || $item.hasClass('js-item-done')) {
+    const hiddenIdEl = el.querySelector(".hiddenId");
+
+    if (
+      hiddenIdEl === null ||
+      el.classList.contains("js-item-pending") ||
+      el.classList.contains("js-item-done")
+    ) {
       return;
     }
-    if ($item.find('.hiddenId').length === 0) {
-      return;
+
+    let handlerName = "ils";
+    if (el.dataset.handlerName) {
+      handlerName = el.dataset.handlerName;
+    } else {
+      const handlerNameEl = el.querySelector(".handler-name");
+
+      if (handlerNameEl !== null) {
+        handlerName = handlerNameEl.value;
+      }
     }
-    var id = $item.find('.hiddenId').val();
-    var handlerName = 'ils';
-    if ($item.data("handler-name")) {
-      handlerName = $item.data("handler-name");
-    } else if ($item.find('.handler-name').length > 0) {
-      handlerName = $item.find('.handler-name').val();
+
+    // update element to reflect lookup
+    el.classList.add("js-item-pending");
+    el.classList.remove("hidden");
+    el.querySelector(".callnumAndLocation").classList.remove("hidden");
+    el.querySelector(".callnumAndLocation .ajax-availability").classList.remove("hidden");
+    const statusEl = el.querySelector(".status");
+    if (statusEl) {
+      statusEl.classList.remove("hidden");
     }
 
-    //queue the element into the handler
-    checkItemHandlers[handlerName].itemQueueAjax(id, $item);
+    //queue the element into the queue
+    checkItemHandlers[handlerName].add({ el, id: hiddenIdEl.value });
   }
 
-  function checkItemStatuses(_container) {
-    var container = typeof _container === 'undefined'
-      ? document.body
-      : _container;
-
-    var ajaxItems = $(container).find('.ajaxItem');
-    for (var i = 0; i < ajaxItems.length; i++) {
-      checkItemStatus($(ajaxItems[i]));
-    }
+  function checkAllItemStatuses(container = document) {
+    container.querySelectorAll(".ajaxItem").forEach(checkItemStatus);
   }
-  function init(_container) {
-    var container = typeof _container === 'undefined'
-      ? document.body
-      : _container;
+
+  function init($container = document) {
+    const container = $container instanceof Node ? $container : $container[0];
```

Comment:
Was just thinking should there be a global function, which would check if the object is instanceof jquery and return the 0 index element if true. It could make the transition from jquery to native js a littlebit smoother and replacing them later would be nicer?